lib/processor.js

```diff
@@ -165,42 +163,42 @@ module.exports = function(config, renderer) {
 			}
 		},
 
-		proxy: function(req, res) {
-			// jscs:disable disallowDanglingUnderscores
+		proxy: function (req, res) {
 			req.headers = req.headers || {};
 
 			var host = req.headers.host || '';
 			var route = router.map(host, req.url);
 			var err = null;
 
-			var rewriteRequestHeaders = function() {
+			var rewriteRequestHeaders = function () {
 				if (route.changeOrigin) {
 					req.headers['X-Orig-Host'] = host;
 				}
 			};
 
-			if (!route) {
-				err = new Error('Request did not match any route');
-				err.status = 404;
-				dispatch.send(err, null, req, res);
-			} else if (!route.host) {
-				err = new Error('Route does not define a host');
-				err.status = 500;
-				dispatch.send(err, null, req, res);
-			} else {
-				route.target = 'http://' + route.host + (route.port ? ':' + route.port : '');
+			if (route) {
+				if (route.host) {
```

Comment:
Again, this refactor seems worse.  If there is a rule banning `if(!x)` in XO then I don't think it is a good thing.  This is not a subjective/presentational opinion - it's going to stop us improving the readability of code.